lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosResult.cxx

```diff
@@ -271,6 +277,138 @@ Function FunctionalChaosResult::getComposedMetaModel() const
+
+  // Create the active transformation and its inverse
+  const Distribution measureMarginal(orthogonalBasisMarginal.getMeasure());
+  const DistributionTransformation transformationMarginal(inputDistributionMarginal, measureMarginal);
+  const DistributionTransformation inverseTransformationMarginal(transformationMarginal.inverse());
+
+  // Condition the multi-indices (taking into account for model selection)
+  // Get the indices of active multi-indices in the reduced enumeration rule
+  Indices listOfActiveReducedIndices(0);  // In the reduced enumeration rule
+  Indices listOfActiveIndices(0);  // In the list of coefficients
+  const UnsignedInteger numberOfFunctions = Psi_k_.getSize();
+  for (UnsignedInteger k = 0; k < numberOfFunctions; ++k)
+  {
+    const Function basisFunction(Psi_k_[k]);
+    const Indices multiIndex(enumerateFunction(I_[k]));
+    // See if this function has active marginal indices only
+    bool isActive = true;
+    for (UnsignedInteger i = 0; i < inputDimension; ++i)
+      if (!(conditioningIndices.contains(i)) && multiIndex[i] > 0)
+      {
+        isActive = false;
+        break;
+      }
+    if (isActive)
+    {
+      listOfActiveIndices.add(k);
+      Indices activeMultiIndex(activeDimension);
+      UnsignedInteger localIndex = 0;
+      for (UnsignedInteger i = 0; i < inputDimension; ++i)
+        if (conditioningIndices.contains(i))
+          activeMultiIndex[localIndex] = multiIndex[i];
+      UnsignedInteger activeIndice = enumerateFunctionMarginal.inverse(activeMultiIndex);
+      listOfActiveReducedIndices.add(activeIndice);
+    }
+  }
+  const UnsignedInteger reducedActiveBasisDimension = listOfActiveReducedIndices.getSize();
+
+  // Compute active coefficients
+  const UnsignedInteger outputDimension = outputSample_.getDimension();
+  Sample activeCoefficients(reducedActiveBasisDimension, outputDimension);
+  const UnsignedInteger numberOfActiveIndices = listOfActiveIndices.getSize();
+  for (UnsignedInteger k = 0; k < numberOfActiveIndices; ++k)
+    activeCoefficients[k] = alpha_k_[listOfActiveIndices[k]];
+
+  // Get the conditioned functional basis
+  Basis activeReducedBasis(0);
+  for (UnsignedInteger k = 0; k < reducedActiveBasisDimension; ++k)
+  {
+    const UnsignedInteger activeIndice = listOfActiveReducedIndices[k];
+    const Function activeFunction(orthogonalBasisMarginal.build(activeIndice));
+    activeReducedBasis.add(activeFunction);
+  }
+
+  // Create the conditional expectation PCE
+  const FunctionalChaosResult conditionalPCE(
+      inputSampleMarginal,
+      outputSample_,
+      inputDistributionMarginal,
+      transformationMarginal,
+      inverseTransformationMarginal,
+      orthogonalBasisMarginal,
+      listOfActiveReducedIndices,
+      activeCoefficients,
+      activeReducedBasis,
+      residuals_,
+      relativeErrors_
+  );
+  return conditionalPCE;
```

Comment:
Perhaps the content is confusing because the example presents both methods. The `getConditionalExpectation` method returns the $\textrm{ce}$ function defined as:

$$
\textrm{ce}(\boldsymbol{x}_u) = \mathbb{E}( \textrm{PCE}(X) | \boldsymbol{X}_u=\boldsymbol{x}_u)
$$

where $\textrm{PCE}(X)$ is the polynomial chaos expansion of the physical model $g$. It happens that the conditional expectation of a PCE is a PCE: isn'it nice? This is why the method returns a `FunctionalChaosResult`. We can evaluate this conditional PCE at any arbitrary point $\boldsymbol{x}_u$.

So let me clarify some confusing points in your statement.
- _"getConditionalExpectation does not actually gives the expectation vector E(Y|X_u=x_u)"_. Indeed, we do not consider the conditional expectation of the _exact_ $Y$ but only the conditional expectation _of its PCE decomposition_. 
- _"but the full chaos decomposition of Y|X_u"_ I do not agree on this point: the method returns the conditional expectation of the PCE.
- _"the doc does not mention that the conditioning values are the E(X_u) mean values"_ Indeed: this is because the method does _not_ set the conditioning values to the mean values.
- In the example, we create a parametric function where the marginals complementary to the set $u$ are fixed to their mean values. But this is only to show the difference with the conditional expectation returned by `getConditionalExpectation`.
- _"unlike your graph."_ The graph compares two methods: the one obtained from `getConditionalExpectation` and the parametric function that we get when we set some parameters to their expected values.
- Notice that, in the example, we do not set $x_u$ to $\mathbb{E}(X_u)$ (as you mentioned), but we set $x_v$ to $\mathbb{E}(X_v)$ where $v$ is the complement of $u$.

In 1ef11943e89aa141eaa43b6226938fd62d846bef, I tried to improve the doc.

Let me know if my statement of the method is clear enough or if there is any confusion still in the doc or example.

Reply:
ok, I didnt got it the first time